--- ypbind-mt-1.11/src/ypbind-mt.c.broadcast	Tue Jun 11 06:50:48 2002
+++ ypbind-mt-1.11/src/ypbind-mt.c	Tue Jun 11 06:54:34 2002
@@ -545,19 +545,9 @@
       exit (1);
     }
 
-  if (!use_broadcast)
-    {
-      if (load_config (0) != 0)
-	{
-	  fputs (_("No NIS server and no -broadcast option specified.\n"), stderr);
-	  fprintf (stderr,
-		   _("Add a NIS server to the %s configuration file,\n"),
-		   configfile);
-	  fputs (_("or start ypbind with the -broadcast option.\n"), stderr);
-	  exit (1);
-	}
-    }
-  else
+  if (!use_broadcast && (load_config (0) != 0))
+    use_broadcast = 1;
+  if (use_broadcast)
     add_server (domain, NULL, 0);
 
   unlink_bindingdir ();
